src/adapter/src/catalog/migrate.rs

```diff
@@ -116,6 +137,39 @@ pub(crate) async fn migrate(
 // AST migrations -- Basic AST -> AST transformations
 // ****************************************************************************
 
+fn ast_rename_avg_function_0_66_0(
+    stmt: &mut mz_sql::ast::Statement<Raw>,
+) -> Result<(), anyhow::Error> {
+    struct AvgFunctionRenamer;
+    impl<'ast> VisitMut<'ast, Raw> for AvgFunctionRenamer {
+        fn visit_function_mut(&mut self, node: &'ast mut Function<Raw>) {
+            // Rewrite the name of the function to be "avg_internal_v1".
+            if let Some(last_component) = node.name.name_mut().0.last_mut() {
```

Comment:
Can this match `user_schema.avg` too?

Reply:
We probably could, but I'm not sure if we're guaranteed to have the schema at this point. Since we don't have user defined functions I think we're okay with not matching on the schema, but I can add the check if you really think we should!

Reply:
Nvm we need to migrate the schema for the function too, so I'll add this check!

Reply:
:+1: i forgot users couldn't create functions